src/v/archival/tests/ntp_archiver_reupload_test.cc

```diff
@@ -310,7 +310,7 @@ FIXTURE_TEST(test_upload_compacted_segments, reupload_fixture) {
     initialize(segments);
```

Comment:
Why does `reupload_fixture` create it's own archiver? Is it in order to avoid waiting for the reconciliation interval of the scheduler?

Reply:
yes, that and it allows us to access the result of the upload process for making assertions, which is otherwise not accessible.